### PR TITLE
[fix](cluster key) Cluster key skip non vertical compact

### PR DESCRIPTION
**Pull request by mymeiyi**
now the cluster key does not support non vertical compaction.
https://github.com/apache/doris/pull/41208 try to implement it, but there are some problems, so we skip non vertical compaction for cluster key

### CODE REVIEW

**Comment by mymeiyi** [comment]
run p0

**Comment by mymeiyi** [comment]
run buildall

**Comment by mymeiyi** [comment]
run buildall

File: be/src/olap/compaction.cpp [hunk 1]
```diff
@@ -186,7 +186,7 @@ Status Compaction::merge_input_rowsets() {
     Status res;
     {
         SCOPED_TIMER(_merge_rowsets_latency_timer);
-        if (_is_vertical) {
+        if (_is_vertical || !_tablet->tablet_schema()->cluster_key_idxes().empty()) {
```
Comment (dataroaring) [inline be/src/olap/compaction.cpp:189]: if non-vertical is configed and table has cluster key, just skip compaction by return error.

**Comment by mymeiyi** [comment]
run buildall

**Comment by mymeiyi** [comment]
run performance

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by bobhan1 — APPROVED** [review]

**Pull request merged**